### DISCUSSION TITLE
Scale testing

### DISCUSSION THREAD

**Discussion by drmorr0** [acrlabs/simkube#73] [Feature Requests and Future Work]
We'd like to know how "big" we can actually go with our simulation.  I've done ~1000 nodes on my local laptop, but sortof by accident, I haven't done any serious benchmarking or profiling.  We'd also like to validate how much space traces _actually_ take up.  It would additionally be cool to do some kind of "before/after" scale testing with both the VirtualKubelet implementation and KWOK (see #70).

**Comment by Huang-Wei** [comment]
According to my conversation with @wzshiming, if the VK impl. sticks to Informer, it has to keep all API objects in memory, and that'd consume a large amount of memory.

In the live demo in KubeCon, at the end of 5k nodes / 10 pods scheduling process, the memory usage of kwok is ~200Mi. (observed from macOS's activity monitor)

Would love to see the comparison.
